test/sequential/test-buffer-bad-overload.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+var common = require('../common');
+var assert = require('assert');
+
+try {
+  new Buffer(10);
+} catch (err) {
+  assert.fail('should not throw');
+}
+
+try {
+  new Buffer(10, 'hex');
+} catch (err) {
+  assert.ok('should throw');
```

Comment:
asserts in `catch` block are kind of useless as they are not evaluated if nothing is thrown. Also, this will never throw anyway.